component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -170,6 +197,50 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
         }
     }
 
+    /**
+     * get String that matches UsernameRegex from subjectDN.
+     *
+     * @param certAttributes        certificate x500 principal
+     * @param authenticationContext authentication context
+     * @throws AuthenticationFailedException
+     */
+    private String getMatchedSubjectAttribute(String certAttributes, AuthenticationContext authenticationContext)
+            throws AuthenticationFailedException {
+
+        LdapName ldapDN;
+        try {
+            ldapDN = new LdapName(certAttributes);
+        } catch (InvalidNameException e) {
+            throw new AuthenticationFailedException("error occurred while get the certificate claims", e);
+        }
+        String userNameAttribute = getAuthenticatorConfig().getParameterMap().get(X509CertificateConstants.USERNAME);
+        List<String> matchedStringList = new ArrayList<>();
+        for (Rdn distinguishNames : ldapDN.getRdns()) {
+            if (subjectPatternCompiled != null && userNameAttribute.equals(distinguishNames.getType())) {
+                Matcher m = subjectPatternCompiled.matcher(String.valueOf(distinguishNames.getValue()));
+                addMatchStringsToList(m, matchedStringList);
+            }
+        }
+        if (matchedStringList.isEmpty()) {
+            authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                    X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR_CODE);
+            log.debug(X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR);
+            throw new AuthenticationFailedException(X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR);
+        } else if (matchedStringList.size() > 1) {
+            authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                    X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_MULTIPLE_MATCHES_ERROR_CODE);
+            log.debug("More than one value matched with the given regex, matches: " + Arrays.toString(matchedStringList.toArray()));
```

Comment:
Add a condition to check whether the debug is enabled or not before printing debug logs and check this throughout the PR

Reply:
Fixed with this PR https://github.com/wso2-extensions/identity-outbound-auth-x509/pull/30/files#diff-3a5508e707adadd20b2b4e9c46f81b90R229